### PR TITLE
ImagelistField uses com_media extension

### PR DESCRIPTION
**Pull request by NicolasDerumigny**
Pull Request for Issue #39936  .

### Summary of Changes
`ImageListField` fetches its list of valid extensions from the `com_media` valid image extension preference.

### Testing Instructions
Add a new `ImageListField` and set it to a directory containing WEBP images, or any new type defined in the media manager valid image extensions (such as AVIF).

### Actual result BEFORE applying this Pull Request
WEBP images are not shown in the list field.

### Expected result AFTER applying this Pull Request
WEBP images are shown in the list field.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
Please revert coding style changes.

**Comment by NicolasDerumigny** [comment]
Which ones? I passed the files to `phpcbf` in order to comply with the integration tests ; what is the correct formatting here?

**Comment by Quy** [comment]
https://github.com/joomla/joomla-cms/pull/39942/files

See unrelated changes such as single quote to double quote and removing spaces.

File: libraries/src/Form/Field/ImagelistField.php [hunk 1]
```diff
@@ -3,43 +3,61 @@
 /**
  * Joomla! Content Management System
  *
- * @copyright  (C) 2010 Open Source Matters, Inc. <https://www.joomla.org>
- * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright (C) 2010 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license   GNU General Public License version 2 or later; see LICENSE.txt
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
 * @copyright  (C) 2010 Open Source Matters, Inc. <https://www.joomla.org>
 * @license    GNU General Public License version 2 or later; see LICENSE.txt
```
Please revert your change or accept my suggestion which does the same. We require 2 spaces between the longest tag and the text.

File: libraries/src/Form/Field/ImagelistField.php [hunk 2]
```diff
@@ -3,43 +3,61 @@
...
  */
 
 namespace Joomla\CMS\Form\Field;
 
+use Joomla\CMS\Component\ComponentHelper;
+
 // phpcs:disable PSR1.Files.SideEffects
-\defined('JPATH_PLATFORM') or die;
+\defined("JPATH_PLATFORM") or die();
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
\defined('JPATH_PLATFORM') or die;
```
Please revert your change or accept my suggestion which does the same. You should not make unrelated changes to code style.

File: libraries/src/Form/Field/ImagelistField.php [hunk 3]
```diff
@@ -3,43 +3,61 @@
...
 // phpcs:enable PSR1.Files.SideEffects
 
 /**
  * Supports an HTML select list of image
  *
- * @since  1.7.0
+ * @since 1.7.0
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
 * @since  1.7.0
```
Please revert your change or accept my suggestion which does the same. We require 2 spaces between the tag and the text.

File: libraries/src/Form/Field/ImagelistField.php [hunk 4]
```diff
@@ -3,43 +3,61 @@
...
  */
 class ImagelistField extends FilelistField
 {
     /**
      * The form field type.
      *
-     * @var    string
-     * @since  1.7.0
+     * @var   string
+     * @since 1.7.0
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
     * @var    string
     * @since  1.7.0
```
Please revert your change or accept my suggestion which does the same. We require 2 spaces between the longest tag and the text.

File: libraries/src/Form/Field/ImagelistField.php [hunk 5]
```diff
@@ -3,43 +3,61 @@
...
      */
-    protected $type = 'Imagelist';
+    protected $type = "Imagelist";
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
    protected $type = 'Imagelist';
```
Please revert your change or accept my suggestion which does the same. We require single quotes to be used whenever possible.

File: libraries/src/Form/Field/ImagelistField.php [hunk 6]
```diff
@@ -3,43 +3,61 @@
...
 
     /**
      * Method to get the list of images field options.
      * Use the filter attribute to specify allowable file extensions.
      *
-     * @return  array  The field option objects.
+     * @return array  The field option objects.
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
     * @return  array  The field option objects.
```
Please revert your change or accept my suggestion which does the same. We require 2 spaces between the longest tag and the text.

File: libraries/src/Form/Field/ImagelistField.php [hunk 7]
```diff
@@ -3,43 +3,61 @@
...
      *
-     * @since   1.7.0
+     * @since 1.7.0
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
     * @since   1.7.0
```
Please revert your change or accept my suggestion which does the same.

File: libraries/src/Form/Field/ImagelistField.php [hunk 8]
```diff
@@ -3,43 +3,61 @@
...
+                "|",
+                array_map(
+                    function ($extension) {
+                        return "\." . $extension . '$';
+                    },
+                    explode(
+                        ",",
+                        ComponentHelper::getParams("com_media")->get(
+                            "image_extensions",
+                            "bmp,gif,jpg,jpeg,png,webp,avif"
```
Comment (richard67) [inline libraries/src/Form/Field/ImagelistField.php]:
```suggestion
                '|',
                array_map(
                    function ($extension) {
                        return '\.' . $extension . '$';
                    },
                    explode(
                        ',',
                        ComponentHelper::getParams('com_media')->get(
                            'image_extensions',
                            'bmp,gif,jpg,jpeg,png,webp,avif'
```
Consistently use single quotes.

**Comment by richard67** [comment]
> https://github.com/joomla/joomla-cms/pull/39942/files
> 
> See unrelated changes such as single quote to double quote and removing spaces.

@NicolasDerumigny To help you with that I have made review suggestions on GitHub. Just to go the link provided in the above comment. You can add my suggestions to a batch and then commit them, or if that is easier for you just accept it one by one. Thanks in advance.

**Comment by richard67** [comment]
> Which ones? I passed the files to `phpcbf` in order to comply with the integration tests ; what is the correct formatting here?

@NicolasDerumigny We use the command here in our integration test:
https://github.com/joomla/joomla-cms/blob/4.2-dev/.drone.yml#L22

```
./libraries/vendor/bin/phpcs --extensions=php -p --standard=ruleset.xml .
```
It requires `composer install` having run before on a git clone in order to get the necessary tools.

**Comment by NicolasDerumigny** [comment]
Thank you for the clarification. I will use it if I need to PR again. I have accepted all your changes, thanks a lot!

**Comment by joomdonation** [comment]
@NicolasDerumigny Look like we have it fixed in 4.3-dev, please see https://github.com/joomla/joomla-cms/blob/4.3-dev/libraries/src/Form/Field/ImagelistField.php#L43 . Maybe we can just backport that change to 4.2-dev ?

(It's simpler. We just list all kind of images, not depend on the settings from Media Manager)

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.3-dev.

**Comment by obuisard** [comment]
We have a correction from https://github.com/joomla/joomla-cms/blob/4.3-dev/libraries/src/Form/Field/ImagelistField.php#L43 therefore I am closing this PR.
Thank you Nicolas @NicolasDerumigny for the PR and everybody for the help provided.
